### PR TITLE
update nlc examples to use new method signatures

### LINKED ISSUE #307: is create_classifier for NLC broken?
I'm following the API docs at [1] to create an NLC classifier

```python
with open('../resources/weather_data_train.csv', 'rb') as training_data:
  classifier = natural_language_classifier.create(
    training_data=training_data,
    name='My Classfier',
    language='en'
  )
print(json.dumps(classifier, indent=2))
```

Which is out of date given the current method name and signature is:

```python
    def create_classifier(self,
                          metadata,
                          training_data,
                          metadata_filename=None,
                          training_data_filename=None):
```

Anyway, I'm calling it this way:

```python
    natural_language_classifier = NaturalLanguageClassifierV1(
        username="keyfrombluemix",
        password="keyfrombluemix"
    )

    with open('weather_data_train.csv', 'rb') as training_data:
        metadata = {"name": "My Classifier", "language": "en"}
        classifier = natural_language_classifier.create_classifier(
            metadata=metadata,
            training_data=training_data
        )
    print(json.dumps(classifier, indent=2))
```

And I end up with a stack trace:

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "welcome.py", line 40, in Welcome
    with open('weather_data_train.csv', 'rb') as training_data:
  File "/usr/local/lib/python2.7/site-packages/watson_developer_cloud/natural_language_classifier_v1.py", line 134, in create_classifier
    accept_json=True)
  File "/usr/local/lib/python2.7/site-packages/watson_developer_cloud/watson_service.py", line 382, in request
    **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 474, in request
    prep = self.prepare_request(req)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 407, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 305, in prepare
    self.prepare_body(data, files, json)
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 499, in prepare_body
    (body, content_type) = self._encode_files(files, data)
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 158, in _encode_files
    fdata = fp.read()
AttributeError: 'dict' object has no attribute 'read'
```

Looking at the source code, I'm a bit confused since it's asking for the metadata as a JSON and then the metadata filename (which there isn't one). 

```
        :param file metadata: Metadata in JSON format. The metadata identifies the language of the data, and an optional name to identify the classifier. For details, see the [API reference](https://www.ibm.com/watson/developercloud/natural-language-classifier/api/v1/#create_classifier).
        :param file training_data: Training data in CSV format. Each text value must have at least one class. The data can include up to 15,000 records. For details, see [Using your own data](https://www.ibm.com/watson/developercloud/doc/natural-language-classifier/using-your-data.html).
        :param str metadata_filename: The filename for training_metadata.
        :param str training_data_filename: The filename for training_data.
```

To make matters worse, the example has been commented out for two years: https://github.com/watson-developer-cloud/python-sdk/blob/master/examples/natural_language_classifier_v1.py#L18-L22

[1] https://www.ibm.com/watson/developercloud/natural-language-classifier/api/v1/?python#create_classifier

### PR DESCRIPTION
**Pull request by ammardodin**
fixes #307

### CODE REVIEW

File: examples/natural_language_classifier_v1.py [hunk 1]
```diff
@@ -18,8 +18,12 @@
 
 # create a classifier
 # with open('../resources/weather_data_train.csv', 'rb') as training_data:
-#     print(json.dumps(natural_language_classifier.create(
-# training_data=training_data, name='weather'), indent=2))
+#     metadata = json.dumps({'name': 'my-classifier', 'language': 'en'})
+#     classifier = natural_language_classifier.create_classifier(
+#         metadata=metadata,
```
Comment (stevemart) [inline examples/natural_language_classifier_v1.py]: this should be training_metadata now, right?

File: examples/natural_language_classifier_v1.py [hunk 2]
```diff
@@ -18,8 +18,12 @@
 
 # create a classifier
```
Comment (stevemart) [inline examples/natural_language_classifier_v1.py]: is there a reason why these are commented out? are the examples run with some sort of CI?

File: examples/natural_language_classifier_v1.py (hunk 1 above)
Comment (ammardodin) [reply examples/natural_language_classifier_v1.py]: `training_metadata` is what we name the parameter when we send it to the service, but the parameter is still called `metadata` for your purposes.

File: examples/natural_language_classifier_v1.py (hunk 2 above)
Comment (ammardodin) [reply examples/natural_language_classifier_v1.py]: @germanattanasio ☝️

File: examples/natural_language_classifier_v1.py (hunk 1 above)
Comment (stevemart) [reply examples/natural_language_classifier_v1.py]: gotcha, cool cool

File: examples/natural_language_classifier_v1.py (hunk 2 above)
Comment (mkistler) [reply examples/natural_language_classifier_v1.py]: Yes, the examples serve double-duty as they are used as integration tests.

**Pull request merged**